Feature 230 warnings

bikegeek opened the pull request:
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>
- testing done on MacOS, Linux ('seneca' and 'eyewall')
- using clone METcalcpy code, feature_153_deprecation_warnings branch 
- using clone METplotpy code, feature_230_warnings branch
- in METplotpy/test directory, ran the following tests and verified no more Deprecation, Runtime, or FutureWarning messages:
    - test_bar.py
    - test_box.py
    - test_difficulty.py
    - test_eclv.py
    - test_ens_ss.py
    - test_equivalence_testing_bounds.py
    - test_histogram.py
    - test_prob_hist.py
    - test_rank_hist.py
    - test_rel_hist.py
    - test_hovmoeller.py
    - test_import.py
    - test_line_group.py
    - test_line_plot.py
    - test_mpr_plot.py
    - test_performance_diagram.py
    - test_reliability_diagram.py
    - test_roc_diagram.py
    - test_taylor_diagram.py
    - test_wind_rose.py

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
   - on 'seneca', activate the /home/met_test/.conda/envs/metplotpy_py38_6 environment
   -   clone the METcalcpy feature_153_deprecation_warnings code
   -   clone the METplotpy feature_230_warnings code
   -   set the PYTHONPATH $HOME/feature_230/METcalcpy:$HOME/feature_230/METplotpy
   -  run these tests in the METplotpy/test directory:
      - test_bar.py
      - test_box.py
      - test_eclv.py
      - test_ens_ss.py
      - test_equivalence_testing_bounds.py
      - test_import.py
      - test_line_plot.py
      - test_line_groups_plot.py
      - test_mpr_plot.py
      - test_reliability.py

and verify that there aren't any DeprecationWarning, FutureWarning, or RuntimeWarning messages generated when running the tests.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[NA]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[ No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **before beta1**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [x] Close the linked issue and delete your feature or bugfix branch from GitHub.

bikegeek committed:
issue #230 to ignore numpy All-NaN slice RuntiimeWarning for tests involving the PCT and CTC line types

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METplotpy into feature_230_warnings

bikegeek committed:
Issue #230 add RuntimeWarning to simplefilter to suppress third-party warnings

bikegeek committed:
Issue #230 add DeprecationWarning to simple filter to suppress third-party warning and replace series.append with pandas.concat to address pandas deprecation

bikegeek committed:
Issue #230 address pandas deprecation by using pd.concat rather than series.append; also removed unused imports

bikegeek committed:
Issue #230 replace Series.append with pd.concat to address pandas deprecation

bikegeek committed:
Issue #230 replace Series.append with pd.concat to address pandas FutureWarning

bikegeek committed:
Issue #230 move order of tests to avoid strange artifact with RuntimeWarning numpy.ndarray size changed

bikegeek committed:
Issue #230 turn multiple keys to explicit list in groupby to address Pandas FutureWarning

bikegeek committed:
remove unused env and out of date comments

bikegeek committed:
issue #230 ignore the RuntimeWarning from pandas and numpy when a NaN value is encountered with nanmean, nanmedian, etc. It is supposed to ignore when a NaN value is encountered, not send a RuntimeWarning

TatianaBurek commented:
I pun some tests on dakota and got "RuntimeWarning":
```
export PYTHONPATH=/d3/projects/METViewer/METplotpy/:/d3/projects/METViewer/METcalcpy
export METPLOTPY_BASE=/d3/projects/METViewer/METplotpy/
/d3/projects/METViewer/METviewer_py3.8.6//bin/python /d3/projects/METViewer/METplotpy/metplotpy/plots/line/line.py /d3/projects/METViewer/auto_test/METviewerTest/develop/HEAD/output/data//mtd_ratio.yaml
```
Here are the messages:
```
/d3/projects/METViewer/METviewer_py3.8.6/lib/python3.8/site-packages/numpy/lib/nanfunctions.py:1218: RuntimeWarning: All-NaN slice encountered
  r, k = function_base._ureduce(a, func=_nanmedian, axis=axis, out=out,
/d3/projects/METViewer/METviewer_py3.8.6/lib/python3.8/site-packages/numpy/lib/nanfunctions.py:1218: RuntimeWarning: All-NaN slice encountered
  r, k = function_base._ureduce(a, func=_nanmedian, axis=axis, out=out,........
```

bikegeek committed:
Delete pytest.ini

Remove this and put warnings in the line_series.py code, where the RuntimeWarning is getting generated from numpy.

bikegeek committed:
#230 Remove the warning filters, they go in the line_series.py code

bikegeek committed:
#230 catch the All-NaN RuntimeWarnings in the code that calls the nanfunctions

bikegeek committed:
issue #230 use context manager to catch the All-NaN slice RuntimeWarning where the nanmean/nanmedian/nanfunction code is getting called

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
remove outdated comments

bikegeek committed:
reinstated the import for pytest, which got accidentally removed when removing outdated comments

TatianaBurek reviewed (approved):
Tested with METviewer regression testing and METviewer UI.
Did not see any RuntimeWarning messages

hankenstein2 reviewed (approved):
Tested on seneca, everything looks good.
